[Frontend] Broken test; useTrade uses mock always; pervasive any types in WebSocket hook

## Summary

Three high-severity and four medium-severity frontend issues found during audit.

## Finding F-01: Failing Test — Wrong Fixture Type (HIGH)

**File:** `web/dashboard/__tests__/useTradeData.test.tsx` line 78

Test `'returns trade array from real API when USE_MOCK_DATA is false'` fails:
```
AssertionError: expected undefined to be 45045
```
Fixture is a `Trade` (camelCase `entryPrice`) but `mapApiTrade` reads `raw.price` (snake_case). Fix: use `RawApiTrade` fixture with `price: 45045`.

## Finding F-02: useTrade(id) Always Uses Mock Data (HIGH)

**File:** `web/dashboard/hooks/useTradeData.ts` lines 121-135

```typescript
queryFn: async () => {
    const trades = getMockTrades()  // NO USE_MOCK_DATA guard — always mock
    const trade = trades.find(t => t.id === id)
```

Never calls `apiClient.getTrade(id)`. Production will always show mock data for trade detail.

## Finding F-03: Pervasive any Types in WebSocket Hook (HIGH)

**File:** `web/dashboard/hooks/useWebSocket.ts` lines 153, 191-195

```typescript
const [trades, setTrades] = useState<Record<string, any>[]>([])
const [positions, setPositions] = useState<Record<string, any>[]>([])
const [orders, setOrders] = useState<Record<string, any>[]>([])
const [agents, setAgents] = useState<Record<string, any>[]>([])
```

All WebSocket state is untyped. Server shape mismatches silently accepted.

**Fix:** Use domain types: `Trade[]`, `Position[]`, `Order[]`, `Agent[]`.

## Finding F-04: Silent Mutation Failures (MEDIUM)

**File:** `web/dashboard/hooks/useTradeData.ts` lines 372-437

All 5 trading mutations (`useCreateOrder`, `useStartTrading`, etc.) lack `onError` callbacks. Failed trades are silently swallowed.

## Finding F-05: Mock Mode No Runtime Warning (MEDIUM)

**File:** `web/dashboard/lib/mock-data.ts` line 11

No warning when `USE_MOCK_DATA=true` in non-dev environment. Silent production data bypass.

## Finding F-06: Test Coverage — 6 Pages Have Zero Tests (LOW)

Pages with no tests: `agents`, `performance`, `polymarket`, `risk`, `settings`, `trades`.

## Run Test to Reproduce

```bash
cd web/dashboard && npm test -- --run
# FAIL __tests__/useTradeData.test.tsx > useTrades > returns trade array...
```

## Files
- `web/dashboard/__tests__/useTradeData.test.tsx`
- `web/dashboard/hooks/useTradeData.ts`, `useWebSocket.ts`, `usePerformance.ts`
- `web/dashboard/lib/mock-data.ts`

--- a/web/dashboard/__tests__/useTradeData.test.tsx
+++ b/web/dashboard/__tests__/useTradeData.test.tsx
@@ -36,19 +36,23 @@ function makeWrapper() {
   )
 }
 
-const fakeTrade: Trade = {
+// #104: The API returns snake_case JSON (RawApiTrade shape); the hook
+// runs mapApiTrade to convert to the camelCase Trade type. The mock
+// must provide the raw API shape so the mapping path is exercised.
+const fakeRawTrade = {
   id: 'trade-abc',
+  order_id: 'order-123',
   symbol: 'BTCUSDT',
-  side: 'long',
-  entryPrice: 45045,
-  currentPrice: 45045,
+  exchange: 'binance',
+  side: 'BUY',
+  price: 45045,
   quantity: 0.1,
-  pnl: 0,
-  pnlPercent: 0,
-  agent: 'paper',
-  confidence: 1,
-  timestamp: '2026-03-19T16:00:00Z',
-  status: 'open',
+  quote_quantity: 4504.5,
+  commission: 0.01,
+  commission_asset: 'USDT',
+  executed_at: '2026-03-19T16:00:00Z',
+  is_maker: false,
+  created_at: '2026-03-19T16:00:00Z',
 }
 
 // ── Tests ───────────────────────────────────────────────────────────
@@ -62,7 +66,7 @@ describe('useTrades', () => {
     const { apiClient } = await import('@/lib/api')
     vi.mocked(apiClient.getTrades).mockResolvedValueOnce({
       success: true,
-      data: { trades: [fakeTrade], count: 1 } as unknown as { trades: Trade[]; count: number },
+      data: { trades: [fakeRawTrade], count: 1 } as unknown as { trades: Trade[]; count: number },
       timestamp: new Date().toISOString(),
     })
 

--- a/web/dashboard/__tests__/useTradeData.test.tsx
+++ b/web/dashboard/__tests__/useTradeData.test.tsx
@@ -2,7 +2,7 @@ import { describe, it, expect, vi, beforeEach } from 'vitest'
 import { renderHook, waitFor } from '@testing-library/react'
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 import type { ReactNode } from 'react'
-import { useTrades } from '@/hooks/useTradeData'
+import { useTrades, useTrade } from '@/hooks/useTradeData'
 import type { Trade } from '@/lib/types'
 
 // ── Mock apiClient ──────────────────────────────────────────────────
@@ -125,5 +125,70 @@ describe('useTrades', () => {
     await waitFor(() => expect(vi.mocked(apiClient.getTrades)).toHaveBeenCalled())
 
     expect(apiClient.getTrades).toHaveBeenCalledWith(10, 20)
+  })
+})
+
+// ── useTrade tests ─────────────────────────────────────────────────
+
+describe('useTrade', () => {
+  beforeEach(() => {
+    vi.clearAllMocks()
+  })
+
+  it('returns single trade from real API by id', async () => {
+    const { apiClient } = await import('@/lib/api')
+    vi.mocked(apiClient.getTrades).mockResolvedValueOnce({
+      success: true,
+      data: { trades: [fakeRawTrade], count: 1 } as unknown as { trades: Trade[]; count: number },
+      timestamp: new Date().toISOString(),
+    })
+
+    const { result } = renderHook(() => useTrade('trade-abc'), { wrapper: makeWrapper() })
+
+    await waitFor(() => expect(result.current.isSuccess).toBe(true))
+
+    expect(result.current.data?.data?.id).toBe('trade-abc')
+    expect(result.current.data?.data?.symbol).toBe('BTCUSDT')
+    expect(result.current.data?.data?.entryPrice).toBe(45045)
+    // Should fetch with large limit to avoid truncation
+    expect(apiClient.getTrades).toHaveBeenCalledWith(1000, 0)
+  })
+
+  it('returns null data when trade id not found', async () => {
+    const { apiClient } = await import('@/lib/api')
+    vi.mocked(apiClient.getTrades).mockResolvedValueOnce({
+      success: true,
+      data: { trades: [fakeRawTrade], count: 1 } as unknown as { trades: Trade[]; count: number },
+      timestamp: new Date().toISOString(),
+    })
+
+    const { result } = renderHook(() => useTrade('nonexistent-id'), { wrapper: makeWrapper() })
+
+    await waitFor(() => expect(result.current.isSuccess).toBe(true))
+
+    expect(result.current.data?.data).toBeNull()
+  })
+
+  it('throws when API returns success:false', async () => {
+    const { apiClient } = await import('@/lib/api')
+    vi.mocked(apiClient.getTrades).mockResolvedValueOnce({
+      success: false,
+      data: null as unknown as { trades: Trade[]; count: number },
+      error: 'server error',
+      timestamp: new Date().toISOString(),
+    })
+
+    const { result } = renderHook(() => useTrade('trade-abc'), { wrapper: makeWrapper() })
+
+    await waitFor(() => expect(result.current.isError).toBe(true))
+
+    expect((result.current.error as Error).message).toBe('server error')
+  })
+
+  it('is disabled when id is empty', () => {
+    const { result } = renderHook(() => useTrade(''), { wrapper: makeWrapper() })
+
+    // Should not trigger a fetch — stays in idle/pending state
+    expect(result.current.fetchStatus).toBe('idle')
   })
 })

--- a/web/dashboard/hooks/useTradeData.ts
+++ b/web/dashboard/hooks/useTradeData.ts
@@ -155,8 +155,12 @@ export function useTrade(id: string) {
         timestamp: string
       }>({ queryKey: QUERY_KEYS.trades })
       for (const [, cached] of cachedQueries) {
-        if (!cached) continue
-        const found = cached.data?.find((t: Trade) => t.id === id)
+        // getQueriesData does a prefix match on ['trades'], so it
+        // returns both useTrades entries (data: Trade[]) AND prior
+        // useTrade entries (data: Trade | null). Guard with
+        // Array.isArray so we don't call .find() on a non-array.
+        if (!cached || !Array.isArray(cached.data)) continue
+        const found = cached.data.find((t: Trade) => t.id === id)
         if (found) {
           return {
             success: true as const,
@@ -186,7 +190,12 @@ export function useTrade(id: string) {
       }
     },
     enabled: !!id,
-    staleTime: REFRESH_INTERVALS.trades,
+    // Longer staleTime than the list hook: without refetchInterval
+    // the detail view only refreshes on remount/refocus, so a short
+    // stale window (2s) would go stale immediately and never auto-
+    // refresh, leaving the user with frozen data. 30s gives a
+    // reasonable cache-reuse window for tab switches.
+    staleTime: 30_000,
   })
 }
 

--- a/web/dashboard/hooks/useTradeData.ts
+++ b/web/dashboard/hooks/useTradeData.ts
@@ -119,13 +119,20 @@ export function useTrades(limit = 50, offset = 0) {
 
 // #104: useTrade previously always called getMockTrades(), ignoring
 // USE_MOCK_DATA entirely. Now follows the same mock/real pattern as
-// useTrades. When hitting the real API, fetches the full trade list
-// and filters client-side (no single-trade endpoint exists yet).
+// useTrades. When hitting the real API, checks the useTrades query
+// cache first to avoid a duplicate network request, then falls back
+// to a large-limit fetch (1000) so trades beyond the default page
+// size aren't silently truncated.
 //
 // Returns data: Trade | null. null means the trade wasn't found in
-// the fetched list (not an API error). Callers should check
+// the fetched data (not an API error). Callers should check
 // result.data?.data !== null before rendering detail views.
+//
+// refetchInterval is intentionally omitted: polling a detail view
+// by re-fetching the full trade list on every interval is expensive.
+// staleTime is kept so the cache is reused on tab switches.
 export function useTrade(id: string) {
+  const queryClient = useQueryClient()
   return useQuery({
     queryKey: [...QUERY_KEYS.trades, id],
     queryFn: async () => {
@@ -139,7 +146,29 @@ export function useTrade(id: string) {
         }
       }
 
-      const response = await apiClient.getTrades()
+      // Check the useTrades list cache first to avoid a duplicate
+      // network request when the list and detail views are mounted
+      // simultaneously.
+      const cachedQueries = queryClient.getQueriesData<{
+        success: boolean
+        data: Trade[]
+        timestamp: string
+      }>({ queryKey: QUERY_KEYS.trades })
+      for (const [, cached] of cachedQueries) {
+        if (!cached) continue
+        const found = cached.data?.find((t: Trade) => t.id === id)
+        if (found) {
+          return {
+            success: true as const,
+            data: found,
+            timestamp: cached.timestamp,
+          }
+        }
+      }
+
+      // Cache miss — fetch with a large limit so trades beyond the
+      // default page size (50) aren't silently truncated.
+      const response = await apiClient.getTrades(1000, 0)
       if (!response.success) {
         throw new Error(response.error || 'Failed to fetch trade')
       }
@@ -158,7 +187,6 @@ export function useTrade(id: string) {
     },
     enabled: !!id,
     staleTime: REFRESH_INTERVALS.trades,
-    refetchInterval: REFRESH_INTERVALS.trades,
   })
 }
 

--- a/web/dashboard/hooks/useTradeData.ts
+++ b/web/dashboard/hooks/useTradeData.ts
@@ -121,16 +121,20 @@ export function useTrades(limit = 50, offset = 0) {
 // USE_MOCK_DATA entirely. Now follows the same mock/real pattern as
 // useTrades. When hitting the real API, fetches the full trade list
 // and filters client-side (no single-trade endpoint exists yet).
+//
+// Returns data: Trade | null. null means the trade wasn't found in
+// the fetched list (not an API error). Callers should check
+// result.data?.data !== null before rendering detail views.
 export function useTrade(id: string) {
   return useQuery({
     queryKey: [...QUERY_KEYS.trades, id],
     queryFn: async () => {
       if (USE_MOCK_DATA) {
         const trades = getMockTrades()
-        const trade = trades.find(t => t.id === id)
+        const trade = trades.find(t => t.id === id) ?? null
         return {
           success: true as const,
-          data: trade || null,
+          data: trade,
           timestamp: new Date().toISOString(),
         }
       }
@@ -145,14 +149,16 @@ export function useTrade(id: string) {
         raw && typeof raw === 'object' && 'trades' in raw
           ? (raw as { trades: RawApiTrade[] }).trades.map(mapApiTrade)
           : []
-      const trade = tradeList.find(t => t.id === id)
+      const trade = tradeList.find(t => t.id === id) ?? null
       return {
         success: true as const,
-        data: trade || null,
+        data: trade,
         timestamp: response.timestamp,
       }
     },
     enabled: !!id,
+    staleTime: REFRESH_INTERVALS.trades,
+    refetchInterval: REFRESH_INTERVALS.trades,
   })
 }
 

--- a/web/dashboard/hooks/useTradeData.ts
+++ b/web/dashboard/hooks/useTradeData.ts
@@ -172,7 +172,10 @@ export function useTrade(id: string) {
 
       // Cache miss — fetch with a large limit so trades beyond the
       // default page size (50) aren't silently truncated.
-      const response = await apiClient.getTrades(1000, 0)
+      // TODO (#224): replace with GET /trades/:id once the backend
+      // supports a single-trade endpoint.
+      const MAX_TRADE_LOOKUP_LIMIT = 1000
+      const response = await apiClient.getTrades(MAX_TRADE_LOOKUP_LIMIT, 0)
       if (!response.success) {
         throw new Error(response.error || 'Failed to fetch trade')
       }

--- a/web/dashboard/hooks/useTradeData.ts
+++ b/web/dashboard/hooks/useTradeData.ts
@@ -117,16 +117,39 @@ export function useTrades(limit = 50, offset = 0) {
   })
 }
 
+// #104: useTrade previously always called getMockTrades(), ignoring
+// USE_MOCK_DATA entirely. Now follows the same mock/real pattern as
+// useTrades. When hitting the real API, fetches the full trade list
+// and filters client-side (no single-trade endpoint exists yet).
 export function useTrade(id: string) {
   return useQuery({
     queryKey: [...QUERY_KEYS.trades, id],
     queryFn: async () => {
-      const trades = getMockTrades()
-      const trade = trades.find(t => t.id === id)
+      if (USE_MOCK_DATA) {
+        const trades = getMockTrades()
+        const trade = trades.find(t => t.id === id)
+        return {
+          success: true as const,
+          data: trade || null,
+          timestamp: new Date().toISOString(),
+        }
+      }
+
+      const response = await apiClient.getTrades()
+      if (!response.success) {
+        throw new Error(response.error || 'Failed to fetch trade')
+      }
+
+      const raw: unknown = response.data
+      const tradeList: Trade[] =
+        raw && typeof raw === 'object' && 'trades' in raw
+          ? (raw as { trades: RawApiTrade[] }).trades.map(mapApiTrade)
+          : []
+      const trade = tradeList.find(t => t.id === id)
       return {
         success: true as const,
         data: trade || null,
-        timestamp: new Date().toISOString(),
+        timestamp: response.timestamp,
       }
     },
     enabled: !!id,

--- a/web/dashboard/hooks/useWebSocket.ts
+++ b/web/dashboard/hooks/useWebSocket.ts
@@ -1,7 +1,7 @@
 'use client'
 
 import { useState, useEffect, useRef, useCallback } from 'react'
-import { WebSocketMessage } from '@/lib/types'
+import type { WebSocketMessage, Trade, Position, Order, Agent } from '@/lib/types'
 import { WS_URL, WS_RECONNECT_ATTEMPTS, WS_RECONNECT_INTERVAL } from '@/lib/constants'
 
 export interface WebSocketState {
@@ -156,7 +156,7 @@ export function useWebSocket(options: UseWebSocketOptions = {}) {
     })
   }, [updateState])
 
-  const sendMessage = useCallback((message: any) => {
+  const sendMessage = useCallback((message: Record<string, unknown>) => {
     if (ws.current?.readyState === WebSocket.OPEN) {
       ws.current.send(JSON.stringify(message))
       return true
@@ -194,11 +194,16 @@ export function useWebSocket(options: UseWebSocketOptions = {}) {
 
 // Specialized hook for trading data
 export function useTradingWebSocket() {
-  const [trades, setTrades] = useState<Record<string, any>[]>([])
-  const [positions, setPositions] = useState<Record<string, any>[]>([])
-  const [orders, setOrders] = useState<Record<string, any>[]>([])
-  const [agents, setAgents] = useState<Record<string, any>[]>([])
-  const [marketData, setMarketData] = useState<Record<string, any>>({})
+  // #104: replaced Record<string, any>[] with proper types so
+  // consumers get type-safe access to trade/position/order/agent
+  // fields. WebSocketMessage.data is still `any` at runtime (the
+  // backend sends untyped JSON), but the state arrays document the
+  // expected shape and catch misuse at compile time.
+  const [trades, setTrades] = useState<Trade[]>([])
+  const [positions, setPositions] = useState<Position[]>([])
+  const [orders, setOrders] = useState<Order[]>([])
+  const [agents, setAgents] = useState<Agent[]>([])
+  const [marketData, setMarketData] = useState<Record<string, unknown>>({})
 
   const handleMessage = useCallback((message: WebSocketMessage) => {
     switch (message.type) {

--- a/web/dashboard/hooks/useWebSocket.ts
+++ b/web/dashboard/hooks/useWebSocket.ts
@@ -199,6 +199,16 @@ export function useTradingWebSocket() {
   // fields. WebSocketMessage.data is still `any` at runtime (the
   // backend sends untyped JSON), but the state arrays document the
   // expected shape and catch misuse at compile time.
+  //
+  // WARNING: the backend WS payload uses the same snake_case JSON
+  // encoding as the REST API (encoding/json on Go structs with
+  // snake_case tags). The REST path runs mapApiTrade to convert to
+  // camelCase Trade shape, but the WS path does NOT — message.data
+  // is pushed directly into the Trade[] array. Consumers accessing
+  // e.g. trade.entryPrice will get undefined at runtime because the
+  // wire format has `price`, not `entryPrice`. A mapApiTrade pass
+  // (or a shared mapper per message.type) is needed to close this
+  // gap. Tracked as a follow-up alongside #224.
   const [trades, setTrades] = useState<Trade[]>([])
   const [positions, setPositions] = useState<Position[]>([])
   const [orders, setOrders] = useState<Order[]>([])

